pkg/pod/entrypoint.go

```diff
@@ -223,6 +239,33 @@ func init() {
 	if err != nil {
 		log.Fatalf("failed to marshal replace ready patch bytes: %v", err)
 	}
+
+	cancelAnnotationPath := "/metadata/annotations/" + strings.Replace(cancelAnnotation, "/", "~1", 1)
+	replaceCancelPatchBytes, err = json.Marshal([]jsonpatch.JsonPatchOperation{{
+		Operation: "replace",
+		Path:      cancelAnnotationPath,
+		Value:     cancelAnnotationValue,
+	}})
+	if err != nil {
+		log.Fatalf("failed to marshal replace cancel patch bytes: %v", err)
+	}
+}
+
+// CancelPod updates the Pod's annotations to signal the cancellation
+// by projecting the cancel annotation via the Downward API.
+// Do not wrap the error, return directly, because the caller needs to judge whether it is a not found error
```

Comment:
nit: I prefer not to include details of the calling function in a function docstring. also, it's still possible to inspect error types w/ wrapped errors using errors.is